ibis/backends/bigquery/__init__.py

```diff
@@ -102,6 +123,162 @@ def __init__(self, *args, **kwargs) -> None:
             name, schema=self._session_dataset, database=self.billing_project
         ).op()
 
+    def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
+        self._make_session()
+
+        raw_name = op.name
+
+        project = self.billing_project
+        dataset = self._session_dataset
```

Comment:
Any chance we could use the anonymous dataset, instead? Note, the anonymous dataset is also location-dependent, so we'll need to make sure we get that info when starting a BQ backend.

See: https://github.com/googleapis/python-bigquery-dataframes/pull/194 where BigQuery DataFrames moves away from BQ Session due to a 1 query at a time limitation.

Reply:
This is the anonymous dataset if I understand correctly. I am grabbing the destination from the query that creates a session (in `_make_session()`).

Reply:
Or are you suggesting that we create an anonymous dataset manually, and use that?

Reply:
> This is the anonymous dataset if I understand correctly. I am grabbing the destination from the query that creates a session (in _make_session()).

Technically yes, the BQ Session implementation writes all of it's data to the anonymous dataset.

> Or are you suggesting that we create an anonymous dataset manually, and use that?

Kinda. I'm suggesting we don't use BQ Sessions since it's not actually necessary and has the 1 query at a time limitation, which has been a problem for us in BQ DataFrames. Instead, we can grab the anonymous dataset directly. See: https://github.com/googleapis/python-bigquery-dataframes/pull/158/files#diff-ca8b88da6ee41672b3d377eb32f9b63ab34a24ec18815c144d4667319ceff6f4

Any resources created there have an automatic time-to-live of 24 hours, but that can be adjusted on individual objects. (We are using 7 days for resources we create in BQ DataFrames for now, and I'm considering adding an ability to manually clean up or refresh the expiration).

Reply:
Oh, I think I see now: I simply would avoid passing the session id, but use the anonymous dataset from the now-one-off session created from `SELECT 1`. I can give that a shot.

Reply:
I put up https://github.com/ibis-project/ibis/pull/7548 to remove session use.